### PR TITLE
[Feature] [api] Change spark base api to java 

### PR DESCRIPTION
**Pull request by Kyofin**
## Purpose of this pull request

change spark batch base api to java code.
related to issue :#766 .
base on pr : #742 
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkTransform.java [hunk 1]
```diff
@@ -14,20 +14,28 @@
...
+    }
 
-  override def getConfig: Config = config
+    @Override
+    public Config getConfig() {
+        return config;
+    }
 
-  def process(data: Dataset[Row], env: SparkEnvironment): Dataset[Row];
+    public abstract Dataset<Row> process(Dataset<Row>data, SparkEnvironment env);
```
Comment (wuchunfu) [inline seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkTransform.java]: Please modify `Dataset<Row>data` to `Dataset<Row> data`

File: seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java [hunk 2]
```diff
@@ -14,19 +14,28 @@
...
+    }
 
-  override def getConfig: Config = config
+    @Override
+    public Config getConfig(){
+        return config;
+    }
 
-  def getData(env: SparkEnvironment): Data;
+    public abstract OUT output(Dataset<Row> data, SparkEnvironment env);
```
Comment (wuchunfu) [inline seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java:39]: Why modify input parameters and output types?

Comment (Kyofin) [reply seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java:39]: the origin code is below.
```
  def output(data: Dataset[Row], env: SparkEnvironment): OUT;

```
Could you describe more detial?

Comment (wuchunfu) [reply seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java:39]: > the origin code is below.
> 
> ```
>   def output(data: Dataset[Row], env: SparkEnvironment): OUT;
> ```
> 
> Could you describe more detial?

Am I reading it wrong?

**Comment by Kyofin** [comment]
@CalvinKirs @RickyHuo @garyelephant  review request.

File: seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java (hunk 2 above)
Comment (Kyofin) [reply seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java:39]: Git Record make confuse.
The real  code different is below.
image

**Comment by wuchunfu** [comment]
@huzk8  Thank you very much for your contribution, someone has already submitted and merged this part of the code, please check https://github.com/apache/incubator-seatunnel/pull/1188 and look forward to your next contribution.

**Pull request closed without merging**